src/us/kbase/workspace/database/Workspace.java

```diff
@@ -1068,129 +1067,96 @@ public UserWorkspaceIDs listWorkspaceIDs(
-		final ByteArrayFileCacheManager dataMan = getDataManager(noData);
-		
-		//this is pretty gross, think about a better api here
-		Map<ObjectIDResolvedWS, Map<SubsetSelection, WorkspaceObjectData.Builder>> stddata = null;
-		Map<ObjectIDResolvedWS, Map<SubsetSelection, WorkspaceObjectData.Builder>> refdata = null;
-		try {
-			stddata = db.getObjects(stdpaths, dataMan, 0,
-					!nullIfInaccessible, false, !nullIfInaccessible);
-			refdata = db.getObjects(refpaths, dataMan, calculateDataSize(stddata),
-					//objects cannot be missing at this stage
-					false, true, true);
-			
-			refpaths.clear();
-			stdpaths.clear();
-			
-			final List<WorkspaceObjectData.Builder> toProc = new ArrayList<>();
-			for (final ObjectIdentifier o: loi) {
-				final WorkspaceObjectData.Builder wodb;
-				final ObjectResolution objres = res.getObjectResolution(o);
-				if (objres.equals(ObjectResolution.INACCESSIBLE)) {
-					wodb = null;
-				} else {
-					final ObjectIDResolvedWS resobj = res.getResolvedObject(o);
-					if (objres.equals(ObjectResolution.NO_PATH)) {
-						if (stddata.containsKey(resobj)) {
-							wodb = stddata.get(resobj).get(o.getSubSet());
-						} else {
-							wodb = null; //object was deleted or missing
-						}
+		final List<WorkspaceObjectData.Builder> toProc = new ArrayList<>();
+		for (final ObjectIdentifier o: loi) {
+			final WorkspaceObjectData.Builder wodb;
+			final ObjectResolution objres = res.getObjectResolution(o);
+			if (objres.equals(ObjectResolution.INACCESSIBLE)) {
+				wodb = null;
+			} else {
+				final ObjectIDResolvedWS resobj = res.getResolvedObject(o);
+				if (objres.equals(ObjectResolution.NO_PATH)) {
+					if (stddata.containsKey(resobj)) {
+						wodb = addSubData(stddata.get(resobj), o.getSubSet());
 					} else {
-						// since was resolved by path, object must exist
-						wodb = WorkspaceObjectData.getBuilder(
-								// need a new builder since there may be many OIs to one OIRWS
-								refdata.get(resobj).get(o.getSubSet()))
-								.withUpdatedReferencePath(res.getReferencePath(o));
+						wodb = null; //object was deleted or missing
 					}
+				} else {
+					// since was resolved by path, object must exist
+					wodb = WorkspaceObjectData.getBuilder(refdata.get(resobj))
+							// need a new builder since there may be many OIs to one OIRWS
+							.withSubsetSelection(o.getSubSet())
+							.withUpdatedReferencePath(res.getReferencePath(o));
 				}
-				toProc.add(wodb);
 			}
-			res = null;
-			refdata.clear();
-			stddata.clear();
-			removeInaccessibleDataCopyReferences(user, toProc);
-			// TODO CODE return optionals instead of nulls
-			return toProc.stream().map(b -> b == null ? null : b.build())
-					.collect(Collectors.toList());
-		} catch (RuntimeException | Error | CorruptWorkspaceDBException |
-				WorkspaceCommunicationException | NoSuchObjectException |
-				TypedObjectExtractionException e) {
-			destroyGetObjectsResources(stddata);
-			destroyGetObjectsResources(refdata);
```

Comment:
Since fetching the data is the last thing that happens in this method and the DB layer data fetching method cleans up after itself if things go wrong, there's no longer any need for this cleanup code